include/rmm/device_buffer.hpp

```diff
@@ -186,35 +161,35 @@ class device_buffer {
    * @throws Nothing
    *
    * @param other The `device_buffer` whose contents will be moved into the
-   * newly constructed one
-   *---------------------------------------------------------------------------**/
+   * newly constructed one.
+   *-------------------------------------------------------------------------**/
   device_buffer(device_buffer&& other) noexcept
       : _data{other._data},
         _size{other._size},
         _capacity{other._capacity},
-        _stream{other._stream},
+        _stream{other.stream()},
         _mr{other._mr} {
     other._data = nullptr;
     other._size = 0;
     other._capacity = 0;
-    other._stream = 0;
+    other.set_stream(0);
   }
 
-  /**---------------------------------------------------------------------------*
+  /**--------------------------------------------------------------------------*
    * @brief Copy assignment operator copies the contents of `other`
    *
-   * @param other The `device_buffer` to copy
-   *---------------------------------------------------------------------------**/
+   * @param other The `device_buffer` to copy.
+   *-------------------------------------------------------------------------**/
```

Comment:
This should just be deleted if we want to require an explicit stream for all copies/allocations.

Reply:
How would one copy a device_buffer then?

Reply:
I looked into this, and it made `device_buffer` too difficult to use. There are multiple places in libcudf that use operator=, and removing it makes most of them clunky. I think it violates the principle of least surprise worse to remove it than to leave it and use `other`'s stream in this case.  So instead, I documented the behavior including a note that if a different stream is needed they should call `set_stream()` on `other` first.